### PR TITLE
StarknetBlocks struct change

### PR DESCRIPTION
**Pull request by mikiw**
## Development related changes

- refactor PR related to https://github.com/0xSpaceShard/starknet-devnet-rs/issues/313

## Checklist:

- [x] Applied formatting - `./scripts/format.sh`
- [x] No linter errors - `./scripts/clippy_check.sh`
- [x] No unused dependencies - `./scripts/check_unused_deps.sh`
- [x] Performed code self-review
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [x] Documented the changes
- [ ] Linked the issues which this PR resolves
- [x] Checked the TODO section in README.md if this PR resolves it
- [x] Updated the tests
- [x] All tests are passing - `cargo test`

### CODE REVIEW

**Comment by mikiw** [comment]
Why with an index map this is needed for tests to pass? The test is wrong or get events? Check this on Wednesday. 

```
    // why this is needed?
    blocks.reverse();
```

**Comment by mikiw** [comment]
The problem was inside get_blocks, filtered_blocks needs to be IndexMap

File: crates/starknet-devnet-core/src/blocks/mod.rs [hunk 1]
```diff
@@ -102,7 +109,7 @@ impl StarknetBlocks {
         to: Option<BlockId>,
     ) -> DevnetResult<Vec<&StarknetBlock>> {
         // used btree map to keep elements in the order of the keys
```
Comment (FabijanC) [inline crates/starknet-devnet-core/src/blocks/mod.rs:111]:
```suggestion
        // used IndexMap to keep elements in the order of the keys
```

**Review by FabijanC — CHANGES_REQUESTED** [review]

File: crates/starknet-devnet-core/src/blocks/mod.rs (hunk 1 above)
Comment (mikiw) [reply crates/starknet-devnet-core/src/blocks/mod.rs]: Fixed in c44b87d57201e532a9aa0730be22f74d5fca4d1b

**Review by FabijanC — APPROVED** [review]

**Review by marioiordanov — APPROVED** [review]

**Pull request merged**